src/net.c

```diff
@@ -827,10 +837,12 @@ static int hyper_check_device_match_mac_addr(const char *device,
 		goto err;
 	}
 
+	free(tmp_mac_addr);
 	close(sock);
 	return 0;
```

Comment:
It would be better to assign ret = 0 and then fall through to return that instead of repeating the free and close lines